packages/contracts/contracts/L1/rollup/CanonicalTransactionChain.sol

```diff
@@ -208,7 +212,7 @@ contract CanonicalTransactionChain is ICanonicalTransactionChain, Lib_AddressRes
             uint40
         )
     {
-        return getQueueLength() - getNextQueueIndex();
+        return uint40(queueElements.length) - nextQueueIndex;
```

Comment:
I'm a bit surprised this is allowed actually. 
Maybe `uint40(queueElements.length - nextQueueIndex);` would be better?

Reply:
I don't think `uint40(queueElements.length - nextQueueIndex);` works because `nextQueueIndex` is a uint40

Reply:
Oh lol.